subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskIntegrationTest.groovy

```diff
@@ -282,6 +282,152 @@ class TestTaskIntegrationTest extends JUnitMultiVersionIntegrationSpec {
         extraArgs << [[], ["--tests", "MyTest"]]
     }
 
+    def "options set prior to setting same test framework will warn and have no effect"() {
+        ignoreWhenJupiter()
```

Comment:
Locally, without this line, this class' tests all pass.  On CI, without it they (rightly) failed.  Strange.